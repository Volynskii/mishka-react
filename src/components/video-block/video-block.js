import './video-block.scss';
import React from "react";
// import {VideoPlayer} from './components/video-player.jsx';

export const VideoBlock = () => {
return <article className="video-block">
    <div className="video-block__video">
        {/*<VideoPlayer/>*/}
    </div>

    <section className="video-block__order">
        <div className="video-block__order__wrapper">
        <div className="video-block__order__badge"/>
       <h1 className="video-block__order__headline"> Процесс
           производства</h1>
        <p className="video-block__order__text">По просьбам наших любимых фолловеров
            мы сняли для вас подробное видео о
            том, как появляются наши товары.</p>
        <button className="video-block__order__button">СДЕЛАТЬ ЗАКАЗ</button>
        </div>
    </section>
</article>
};
