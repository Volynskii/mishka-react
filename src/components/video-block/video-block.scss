@import "../../utilities/variables";

.video-block {
  margin-left: auto;
  margin-right: auto;
  display: flex;
  border: 1px solid;
  width: 1050px;
  height: 455px;
}
.video-block__video {
  border: 1px solid;
  width: 50%;
  height: 100%;
  background-image: url("../../images/Catalog/video-desktop.png");
  background-position: center;
}
.video-block__order {
  display: flex;
  justify-content: flex-start;
  align-items: center;
  height: 100%;
  width: 50%;
  border: 1px solid;
}
.video-block__order__wrapper {
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  width: 350px;
  height: 350px;
  margin-left: 80px;
}
.video-block__order__badge {
  width: 32px;
  height: 22px;
  background-image: url("../../images/Catalog/video-badge.png");
  background-position: center;
}
.video-block__order__headline {
  font-size: 20px;
  line-height: 20px;
  font-weight: bold;
}
.video-block__order__text {
  width: 225px;
  font-size: 15px;
  line-height: 30px;
  margin: 0;
  padding: 0;
}
.video-block__order__button {
  font-size: 17px;
  font-weight: bold;
  background-color: transparent;
  width: 228px;
  border: 2px solid black;
}

