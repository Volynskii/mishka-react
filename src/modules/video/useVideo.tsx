import { useState, useEffect } from "react";

export const useVideo = () => {
    const [isPlaying, setPlaying] = useState(false);

    let timerId: any;

    useEffect(() => {
        return () => clearTimeout(timerId);
    }, [timerId]);

    const handleMouseEnter: () => void = () => {
        timerId = setTimeout(() => {
            setPlaying(true);
        }, 1000);
    };

    const handleMouseLeave: () => void = () => {
        clearTimeout(timerId);
        setPlaying(false);
    };
    return {
        isPlaying,
        handleMouseEnter,
        handleMouseLeave
    };

};


