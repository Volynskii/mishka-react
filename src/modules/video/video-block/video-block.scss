@import "../../../utilities/variables";


/*mobile-mobile-mobile */
@media (min-width: $mobile-width-min) and (max-width: $mobile-width-max) {
  .video-block {
    margin-left: auto;
    margin-right: auto;
    margin-bottom: 50px;
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    //border: 1px solid;
    width: 460px;
  }
  .video-block__video {
    width: 260px;
    height:150px;
  }
  .video-block__order {
    display: flex;
    justify-content: center;
    align-items: flex-start;
    width: 100%;
    //border: 1px solid red;
  }
  .video-block__order__wrapper {
     border: 1px solid #e4e4e4;
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    align-items: center;
    width: 260px;
    min-height: 260px;
   // margin-top: 50px;
   // margin-right: 80px;
   // margin-left: 80px;
    margin-bottom: 50px;
  }
  .video-block__order__headline-container {
    display: flex;
    justify-content: center;
    align-items: center;
    width: 100%;
  }
  .video-block__order__badge {
    width: 50px;
    height: 22px;
    background-image: url("../../../images/Catalog/video-badge.png");
    background-position: left;
    background-repeat: no-repeat;
  }
  .video-block__order__headline {
    width: 130px;
    font-size: 17px;
    line-height: 20px;
    font-weight: bold;
    margin: 0;

  }
  .video-block__order__text {
    width: 190px;
    text-align: left;
    font-size: 13px;
    line-height: 24px;
    margin: 0;
    padding: 0;
  }
  .video-block__order__button {
    display: flex;
    justify-content: center;
    align-items: center;
    text-align: center;
    font-size: 17px;
    color: #444444;
    font-weight: bold;
    background-color: transparent;
    min-width: 190px;
    width: 260px;
    height: 52px;
    border: 2px solid black;
    margin-left: auto;
    margin-right: auto;
  }

};

/*tablet-tablet-tablet */
@media (min-width: $tablet-width-min) and (max-width: $tablet-width-max) {
  .video-block {
    margin-left: auto;
    margin-right: auto;
    margin-top: 50px;
    margin-bottom: 50px;
    display: flex;
flex-direction: column;
    justify-content: flex-start;
    align-items: center;
    //border: 1px solid;
    width: 100%;
  }

  .video-block__video {
    width: 77%;
    height:450px;
  }

  .video-block__order {
    display: flex;
    justify-content: flex-start;
    align-items: flex-start;
    height: 100%;
    width: 100%;
    //border: 1px solid;
  }
  .video-block__order__wrapper {

    border: 2px solid #e3e3e3;
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    align-items: center;
    width: 87%;
    height: 210px;
    margin-top: 50px;
    margin-right: 80px;
    margin-left: 80px;
    margin-bottom: 50px;
  }
  .video-block__order__headline-container {
    display: flex;
    justify-content: flex-start;
    width: 580px;
  }
  .video-block__order__badge {
    width: 65px;
    height: 22px;
    background-image: url("../../../images/Catalog/video-badge.png");
    background-position: left;
    background-repeat: no-repeat;
  }
  .video-block__order__headline {
    font-size: 20px;
    line-height: 20px;
    font-weight: bold;
  }
  .video-block__order__text {
    width: 580px;
    font-size: 15px;
    line-height: 30px;
    margin: 0;
    padding: 0;
  }
  .video-block__order__button {
    display: flex;
    justify-content: center;
    align-items: center;
    text-align: center;
    font-size: 17px;
    color: #444444;
    font-weight: bold;
    background-color: transparent;
    width: 580px;
    height: 52px;
    border: 2px solid black;
    margin-left: auto;
    margin-right: auto;
  }

};

/*desktop-desktop-desktop */
@media (min-width: $desktop-width-min) {
  .video-block {
    margin-left: auto;
    margin-right: auto;
    margin-bottom: 50px;
    display: flex;
    justify-content: space-between;
    align-items: center;
    //border: 1px solid;
    width: 1050px;
    height: 450px;
  }
  .video-block__video {
    //border: 1px solid blue;
    //width: 980px;
    width: 700px;
    height: 400px;
    //background-image: url("../../../images/Catalog/video-desktop.png");
    //background-position: center;
  }
  .video-block__order {
    display: flex;
    justify-content: center;
    align-items: center;
    height: 400px;
    width: 50%;
    border-top: 2px solid #ececee;
    border-bottom: 2px solid #ececee;
    border-right: 2px solid #ececee;
  }

  .video-block__order__wrapper {
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    align-items: center;
    width: 350px;
    height: 350px;
   // margin-left: 80px;
  }

  .video-block__order__headline-container {
    width: 230px;
    height: 100px;
    display: flex;
    flex-direction: column;
    justify-content: space-between;
  }

  .video-block__order__badge {
    width: 32px;
    height: 22px;
    background-image: url("../../../images/Catalog/video-badge.png");
    background-position: center;
  }

  .video-block__order__headline {
    width: 200px;
    font-size: 20px;
    line-height: 20px;
    font-weight: bold;
    margin: 0;
  }

  .video-block__order__text {
    width: 230px;
    font-size: 15px;
    line-height: 30px;
    margin: 0;
    padding: 0;
  }

  .video-block__order__button {
    display: flex;
    justify-content: center;
    align-items: center;
    font-size: 17px;
    font-weight: bold;
    text-align: center;
    color: inherit;
    background-color: transparent;
    width: 230px;
    height: 50px;
    border: 2px solid black;
  }

}
