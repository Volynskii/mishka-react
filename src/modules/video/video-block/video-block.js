import './video-block.scss';
import React, { useState, useRef, useEffect} from "react";

export const VideoBlock = ({width,height,src,poster,isPlaying, onMouseEnter, onMouseLeave}) => {

    const videoRef = useRef(null);

    const [isLoading, setLoading] = useState(true);

    useEffect(() => {
        const { current: video } = videoRef;
        video.oncanplaythrough = () => {
            setLoading(false);
        };

        return () => {
            video.oncanplaythrough = null;
        };
    }, []);

    useEffect(() => {
        const { current: video } = videoRef;
        if (isLoading) {
            return;
        }

        if (isPlaying) {
            video.play();
        } else {
            video.load();
        }
    }, [isPlaying]);

return (
    <article className="video-block">
    <video
        ref={videoRef}
        muted={true}
        width={width}
        height={height}
         src={src}
         poster={poster}
           onMouseEnter={onMouseEnter}
           onMouseLeave={onMouseLeave}


    />

    <section className="video-block__order">
        <div className="video-block__order__wrapper">
        <div className="video-block__order__badge"/>
       <h1 className="video-block__order__headline"> Процесс
           производства</h1>
        <p className="video-block__order__text">По просьбам наших любимых фолловеров
            мы сняли для вас подробное видео о
            том, как появляются наши товары.</p>
        <button className="video-block__order__button">СДЕЛАТЬ ЗАКАЗ</button>
        </div>
    </section>
</article>
)
};
