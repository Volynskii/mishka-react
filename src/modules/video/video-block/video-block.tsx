import './video-block.scss';
import React from "react";

type videoBlockContainerProps = {
src: string;
poster: string;
isPlaying: boolean;
onMouseEnter: () => void;
onMouseLeave: () => void;
width: string;
height: string;
}

export const VideoBlock: React.FC<videoBlockContainerProps> = ({
         src,
         poster,
         isPlaying,
         onMouseEnter,
         onMouseLeave,
         width,
         height}) => {

    const videoRef = React.useRef(null);

    const [isLoading, setLoading] = React.useState(true);

    React.useEffect(() => {
        const { current: video } : { current:any } = videoRef;
        video.oncanplaythrough = () => {
            setLoading(false);
        };

        return () => {
            video.oncanplaythrough = null;
        };
    }, []);

    React.useEffect(() => {
        const { current: video }: {current: any} = videoRef;
        if (isLoading) {
            return;
        }

        if (isPlaying) {
            video.play();
        } else {
            video.load();
        }
    }, [isLoading,isPlaying]);

    return (
        <article className="video-block">
        <video
            className={"video-block__video"}
    ref={videoRef}
    muted={true}
    width={width}
    height={height}
    src={src}
    poster={poster}
    onMouseEnter={onMouseEnter}
    onMouseLeave={onMouseLeave}
    />

    <section className="video-block__order">
    <div className="video-block__order__wrapper">
    <section className="video-block__order__headline-container">
    <div className="video-block__order__badge"/>
    <h1 className="video-block__order__headline"> Процесс
    производства</h1>
    </section>
    <p className="video-block__order__text">По просьбам наших любимых фолловеров
    мы сняли для вас подробное видео о
    том, как появляются наши товары.</p>
    <a  href={`/form`} className="video-block__order__button">СДЕЛАТЬ ЗАКАЗ</a>
    </div>
    </section>
    </article>
)
};
