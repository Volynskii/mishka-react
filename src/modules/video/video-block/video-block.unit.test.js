import React, {useRef} from "react";
import {mount} from "enzyme";
import {Provider} from "react-redux";
import configureStore from "redux-mock-store";
import renderer, {act} from "react-test-renderer";
import * as redux from "react-redux";
import {useVideo} from "../useVideo";
import {VideoBlock} from "./video-block.tsx";

let comments = [
    {
        comment: 'Я молодая мама,' +
            ' но всё равно фотографии' +
            ' не набирали и близко такого ' +
            'количества лайков, как у ' +
            'популярных инстамамочек.' +
            ' В отчаянии, я накупила ' +
            'аксессуаров и игрушек в ' +
            'Мишке, и мои фотографии ' +
            'сразу стали более стильными,' +
            ' а также набирают больше лайков!',
        author: 'Анастасия Красильникова',
        email: '@misssssiskras'
    },
    {
        comment: 'Я молодой парень и мои мои' +
            ' фотографии обычно не набирали' +
            ' большого количества лайков, ' +
            'но как только я купил игрушки ' +
            'разных форматов цветов и форматов' +
            ' количество лайков пошло в гору!' +
            ' Я очень доволен! ',
        author: 'Михаил Соколов!',
        email: 'sokolov@gmail.com'
    },
    {
        comment: 'Я очень доволен данными' +
            ' товарами,товары имеют' +
            ' чрезвычайно высокое качество,' +
            ' ' +
            'заказал себе, друзьям, бабушке,дедушке, папе и маме!',

        author: 'Илья Перевалов!',
        email: 'perevalov@gmail.com'
    }

];

const initialState = {
comments:comments,
};
const mockStore = configureStore();
const store = mockStore(initialState);

let results;
const renderHook = (hook) => {

    function HookWrapper() {
        results = hook();
        return null;
    }
    mount(
        <Provider store={store}>
            <HookWrapper/>
        </Provider>
    );
    return results;
};
// const video = jest.fn();
// const useRefSpy = jest.spyOn(React, 'useRef').mockReturnValueOnce({ current: { video } });

renderHook(useVideo)

// const useDispatchSpy = jest.spyOn(React, 'useRef');
// const mockDispatchFn = jest.fn()
// useDispatchSpy.mockReturnValue(mockDispatchFn);

const initialProps = {
    width:`100%`,
height:'100%',
src:null,
poster: null,
isPlaying:results.isPlaying,
onMouseEnter:results.handleMouseEnter,
onMouseLeave:results.handleMouseLeave

};

const container = mount(
    <Provider store={store}>
        <VideoBlock {...initialProps}
        />
    </Provider>
);

describe('VideoBlock works', () => {

    it('should render VideoBlock Component', ()=> {
        const dom = container;
        const component = renderer.create(dom);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    // it('VideoBlock props work', ()=> {
    //     expect(results.isPlaying).toBe(false)
    //     act(() => {
    //         results.handleMouseEnter();
    //     });
    //     expect(results.isPlaying).toBe(false)
    //     act(() => {
    //         results.handleMouseLeave();
    //     });
    //     expect(results.isPlaying).toBe(false)
    // });
});
