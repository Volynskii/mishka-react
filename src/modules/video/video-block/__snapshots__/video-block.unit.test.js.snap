// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoBlock works should render VideoBlock Component 1`] = `
<article
  className="video-block"
>
  <video
    className="video-block__video"
    height="100%"
    muted={true}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    poster={null}
    src={null}
    width="100%"
  />
  <section
    className="video-block__order"
  >
    <div
      className="video-block__order__wrapper"
    >
      <section
        className="video-block__order__headline-container"
      >
        <div
          className="video-block__order__badge"
        />
        <h1
          className="video-block__order__headline"
        >
           Процесс производства
        </h1>
      </section>
      <p
        className="video-block__order__text"
      >
        По просьбам наших любимых фолловеров мы сняли для вас подробное видео о том, как появляются наши товары.
      </p>
      <a
        className="video-block__order__button"
        href="/form"
      >
        СДЕЛАТЬ ЗАКАЗ
      </a>
    </div>
  </section>
</article>
`;
